pkg/api/v2/reporter/reporter.go

```diff
@@ -103,18 +103,25 @@ func (e ResourceReports) ValidateStrict() error {
 	return errs
 }
 
+// Minimal set of client operations required for reporters.
+type ReporterResourceClient interface {
+	Kind() string
+	Read(namespace, name string, opts clients.ReadOpts) (resources.Resource, error)
+	Write(resource resources.Resource, opts clients.WriteOpts) (resources.Resource, error)
+}
```

Comment:
All lgtm, but I don't have any context on this change

Reply:
The interface is the subset of the functions of the `ResourceClient` interface that are actually used by the reporter.

In Gloo, I will provide a special implementation for the `RateLimitConfig` that uses the `Update*Status` function on the skv2 client to update the status sub-resource (the normal `Update` performed by the internal sk clientset would not work on subresources). With this interface I'll just need to implement the functions that are really needed (no `List`, `Delete`, etc.).